src/reducers/dictionary/courseFocus.ts

```diff
@@ -1,48 +1,56 @@
+import Course from '@/shapes/model/subject/Course';
 import {
   RESET,
   SET_COURSE_FOCUS,
   CLEAR_COURSE_FOCUS,
   SET_REVIEWS,
   UPDATE_REVIEW,
   SET_LECTURES,
-} from '../../actions/dictionary/courseFocus';
+  DictionaryAction,
+} from '@/actions/dictionary/courseFocus';
+import Lecture from '@/shapes/model/subject/Lecture';
+import Review from '@/shapes/model/review/Review';
 
-const initialState = {
+interface CourseFocusState {
+  course: Course | null;
+  lectures: Lecture[] | null;
+  reviews: Review[] | null;
+}
+
+const initialState: CourseFocusState = {
   course: null,
   reviews: null,
   lectures: null,
 };
 
-const courseFocus = (state = initialState, action) => {
+const courseFocus = (state = initialState, action: DictionaryAction) => {
   switch (action.type) {
     case RESET: {
       return initialState;
     }
     case SET_COURSE_FOCUS: {
       const courseChanged = !state.course || state.course.id !== action.course.id;
-      return Object.assign(
-        {},
-        state,
-        {
-          course: action.course,
-        },
-        courseChanged ? { reviews: null, lectures: null } : {},
-      );
+
+      return {
+        ...state,
+        course: action.course,
+        reviews: courseChanged ? null : state.reviews,
+        lectures: courseChanged ? null : state.lectures,
+      };
     }
     case CLEAR_COURSE_FOCUS: {
-      return Object.assign({}, state, {
-        course: null,
-        reviews: null,
-        lectures: null,
-      });
+      return { ...state, course: null, reviews: null, lectures: null };
     }
     case SET_REVIEWS: {
-      return Object.assign({}, state, {
-        reviews: action.reviews,
-      });
+      return { ...state, reviews: action.reviews };
     }
     case UPDATE_REVIEW: {
       const originalReviews = state.reviews;
+
+      if (originalReviews == undefined) {
```

Comment:
`originalReviews`가 null일 때도 조기종료 해줘야 할 듯해서 아래처럼 작성하는 거 추천드립니다!

```suggestion
      if (!originalReviews) {
```